fugue_spark/execution_engine.py

```diff
@@ -466,17 +473,21 @@ def _presort_to_col(_col: str, _asc: bool) -> Any:
         if len(partition_spec.partition_by) == 0:
             if len(_presort.keys()) > 0:
                 d = d.orderBy(
-                    [_presort_to_col(_col, _presort[_col]) for _col, in _presort.keys()]
+                    [_presort_to_col(_col, _presort[_col]) for _col in _presort.keys()]
                 )
             d = d.limit(n)
 
         # If partition exists
         else:
             w = Window.partitionBy([col(x) for x in partition_spec.partition_by])
+
             if len(_presort.keys()) > 0:
                 w = w.orderBy(
-                    [_presort_to_col(_col, _presort[_col]) for _col, in _presort.keys()]
+                    [_presort_to_col(_col, _presort[_col]) for _col in _presort.keys()]
                 )
+            else:
+                # row_number() still needs an orderBy
+                w = w.orderBy(lit(1))
```

Comment:
i thought about it. This is an edge case, without sort, it really doesn't make much sense to use take. So I think this is totally fine.